assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -458,16 +458,30 @@ export default Component.extend({
         messageData.hideUserInfo = true;
       }
     }
-    messageData.expanded = !messageData.deleted_at;
+    this._handleMessageHidingAndExpansion(messageData);
     messageData.messageLookupId = this._generateMessageLookupId(messageData);
-    if (this.targetMessageId && this.targetMessageId === messageData.id) {
-      messageData.expanded = true;
-    }
     const prepared = ChatMessage.create(messageData);
     this.messageLookup[messageData.messageLookupId] = prepared;
     return prepared;
   },
 
+  _handleMessageHidingAndExpansion(messageData) {
```

Comment:
Part of me thinks we shouldn't even send a message to someone if we don't want to show it. I understand it might be more complex and cause perf issues, but thought I would say it :p

It would also avoid having to deal with this client side.

Reply:
Well it's like the `deleted_at` stuff, we want to show the placeholder that can be expanded/collapsed. I was also concerned that if we didn't send the messages through via new-messages in MessageBus that it would cause tracking state sync issues.

Reply:
Just remember this ignored functionality copies what we already have in topics -- if you have ignored a user inside the topic all their posts are collapsed behind a "VIEW HIDDEN" button. I do see what you mean though, I did toss up whether I should somehow blackhole all those new messages in the backend and not send them to the client via MessageBus if you have ignored someone. Maybe we should discuss it a bit more internally?

Reply:
I suppose you could argue since Mute and Ignore are supposed to apply to creating DMs, we should not be showing any of their messages at all inside the DMs, but inside public channels Ignore works the same as topics with VIEW HIDDEN?